### ISSUE TITLE
Parse Error in GB and GBA apps

### ISSUE THREAD

**Issue by fran122157** [EstebanFuentealba/MALVEKE-Flipper-Zero#31]
Hey there!

I'm having some problems using the GB and GBA cartridge reader apps. Both of them throw me a "parse error" when trying to read  some cartridges.

(this is following the app instructions downloaded from [here](https://arc.net/l/quote/sbfipvkh))

Checks:
- Updating flipper zero firmware
- Updating board firmware following the guide
- Check that flipper shows "connected" message when inserting malveke board
- Led lights up
- Switch is in EXP mode
- Reinstalling the apps

Am I missing some really obvious thing in the board connection process? 

Really love the project, just hoping to use the board and dump some saves 

Thanks in advance and please let me know if I can add more info to this issue.

**Comment by EstebanFuentealba** [comment]
Hello, how are you? I have a few questions. 

- Which firmware version are you currently using?
- In https://flipperzero.json.cl/ did you select 2.5.1 ?
- What cartridge have you tried?

Before sending them, I tested them with the [Original FW 0.98.3](https://github.com/flipperdevices/flipperzero-firmware/releases/tag/0.98.3).

I also recently tried with [Momentum FW](https://github.com/Next-Flip/Momentum-Firmware/releases/tag/mntm-001) using apps from the official catalog.

Let me see if there have been any recent changes.

https://github.com/EstebanFuentealba/MALVEKE-Flipper-Zero/assets/442927/eee2dec0-3294-4fb8-a82d-be74c0520c97

**Comment by EstebanFuentealba** [comment]
Here are the faps I'm using. Could you test them? And before testing, could you please reboot?

[malveke_faps.zip](https://github.com/EstebanFuentealba/MALVEKE-Flipper-Zero/files/14822947/malveke_faps.zip)

![reboot (4)](https://github.com/EstebanFuentealba/MALVEKE-Flipper-Zero/assets/442927/9b71f015-f25a-4f22-9a1a-a022a6e94934)

**Comment by fran122157** [comment]
Hola!

Thanks for the quick response, I'm currently running the latest firmware (that's 0.99.1)

Yeap, tried rebooting multiple times before posting the issue. 

The faps you just sent me did the trick though! I can reed GB and GBA carts now. Also dump rom files (which were previously throwing a null pointer exception)

![Screenshot-20240320-205251](https://github.com/EstebanFuentealba/MALVEKE-Flipper-Zero/assets/5866185/08caa230-2088-4f5a-9925-833e21718d85)

**Comment by EstebanFuentealba** [comment]
🤔 And where did you download the previous ones from?

**Comment by EstebanFuentealba** [comment]
I updated the link for the FAPs and added the latest version of the Pokémon app (I & II GEN).


> Here are the faps I'm using. Could you test them? And before testing, could you please reboot?
> 
> [malveke_faps.zip](https://github.com/EstebanFuentealba/MALVEKE-Flipper-Zero/files/14676248/malveke_faps.zip)
> 
> ![reboot (4)](https://private-user-images.githubusercontent.com/442927/314773572-9b71f015-f25a-4f22-9a1a-a022a6e94934.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5ODI3MDUsIm5iZiI6MTcxMDk4MjQwNSwicGF0aCI6Ii80NDI5MjcvMzE0NzczNTcyLTliNzFmMDE1LWYyNWEtNGYyMi05YTFhLWEwMjJhNmU5NDkzNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyMVQwMDUzMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZThlYzQyMzY3NmVkMzQ3ZjNjOTdhYmExM2RjZGE2NGRlMzNhNzVhZDJhNDZlOWFhMDdhYWE3ZDc4MjhlYTlkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.0GxDvy2Id6Dli13OdW3bDlWbs3p5CjlPIHoV1xcbz_I)

**Comment by EstebanFuentealba** [comment]
@fran122157  I'll leave this issue open while I investigate what happened with the official catalog.

**Comment by Str8buggin** [comment]
Hey,
I was also having the same issue with a "parse error" we were speaking through tindie. all of the .fap files that were in the zip all work perfect! thank you for the speedy solution. i am running MNTM-001 and got my applications from lab.flipper.net. just tested the pokemon trading and that issue is also fixed.

**Comment by bison--** [comment]
Hi,

TL;DR: thanks, your FAP files work <3 https://github.com/EstebanFuentealba/MALVEKE-Flipper-Zero/issues/31#issuecomment-2010948941

i had the same "parse error" issue after building the FAPs myself.
Downloaded the apps via lab.flipper.net (didn't knew the page existed, saw it in the previous post) but got the same "parse error".
Only with your supplied files I got it to work (thanks for that <3).

my flipper version: 
![image](https://github.com/EstebanFuentealba/MALVEKE-Flipper-Zero/assets/901400/e5ce38a0-23bd-48bc-bd22-31e9666bc13c)

**Comment by fran122157** [comment]
> 🤔 And where did you download the previous ones from?

Sorry for the delay in the response! I downloaded them from flipper lab as other have mentioned in the thread. Just to clarify

Thanks!

**Comment by EstebanFuentealba** [comment]
The apps are now available and working again in the official catalog.

https://github.com/flipperdevices/flipper-application-catalog/pull/331
